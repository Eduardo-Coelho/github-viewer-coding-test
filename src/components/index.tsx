export * from './commitDetails/CommitDetails';
export * from './commitTable/CommitTable';
export * from './repoSearch/RepoSearch';
export * from './loader/Loader';
export * from './modal/Modal';
