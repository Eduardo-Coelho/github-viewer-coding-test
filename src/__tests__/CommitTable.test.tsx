import React from 'react';
import { render } from '@testing-library/react';
import { CommitTable } from '../components';

test('CommitTable matches snapshot', () => {
  const mockCommits = [
    {
      sha: '123abc',
      commit: {
        message: 'Initial commit',
        author: {
          name: 'John Doe',
          email: 'john@example.com',
          date: '2023-05-21T12:34:56Z',
        },
        committer: {
          name: 'Jane Doe',
          email: 'jane@example.com',
          date: '2023-05-21T12:34:56Z',
        },
        url: 'http://example.com/commit/123abc',
      },
    },
    {
      sha: '456def',
      commit: {
        message: 'Second commit',
        author: {
          name: 'Alice',
          email: 'alice@example.com',
          date: '2020-05-22T12:34:56Z',
        },
        committer: {
          name: 'Bob',
          email: 'bob@example.com',
          date: '2021-05-22T12:34:56Z',
        },
        url: 'http://example.com/commit/456def',
      },
    },
  ];

  const mockOnSelectCommit = jest.fn();

  const { asFragment } = render(
    <CommitTable
      commits={mockCommits as any}
      onSelectCommit={mockOnSelectCommit}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});

test('CommitTable matches snapshot with empty commits list', () => {
  const { asFragment } = render(
    <CommitTable commits={[]} onSelectCommit={() => {}} />,
  );
  expect(asFragment()).toMatchSnapshot();
});
