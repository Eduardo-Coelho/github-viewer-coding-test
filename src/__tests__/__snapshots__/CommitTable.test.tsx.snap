// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitTable matches snapshot 1`] = `
<DocumentFragment>
  <table>
    <thead>
      <tr>
        <th>
          Id
        </th>
        <th>
          Message
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          123abc
        </td>
        <td>
          Initial commit
        </td>
      </tr>
      <tr>
        <td>
          456def
        </td>
        <td>
          Second commit
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`CommitTable matches snapshot with empty commits list 1`] = `
<DocumentFragment>
  <table>
    <thead>
      <tr>
        <th>
          Id
        </th>
        <th>
          Message
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</DocumentFragment>
`;
